Content.Client/Interaction/DragDropHelper.cs

```diff
@@ -63,10 +66,16 @@ private enum DragState : byte
     /// <param name="onEndDrag"><see cref="OnEndDrag"/></param>
     public DragDropHelper(OnBeginDrag onBeginDrag, OnContinueDrag onContinueDrag, OnEndDrag onEndDrag)
     {
-        _inputManager = IoCManager.Resolve<IInputManager>();
+        IoCManager.InjectDependencies(this);
         _onBeginDrag = onBeginDrag;
         _onEndDrag = onEndDrag;
         _onContinueDrag = onContinueDrag;
+        _cfg.OnValueChanged(CCVars.DragDropDeadZone, SetDeadZone, true);
+    }
+
+    ~DragDropHelper()
+    {
+        _cfg.UnsubValueChanged(CCVars.DragDropDeadZone, SetDeadZone);
     }
```

Comment:
@mirrorcult, I'm new to C#. Should I implement `IDisposable` instead of the finalizer?

Reply:
in this case either works imo